### DISCUSSION TITLE
Touchscreen 5“

### DISCUSSION THREAD

**Discussion by Tuxrocks** [AndreWohnsland/CocktailBerry#71] [Need help] [answered]
Hallo,
Beim Start von Cocktailberry funktioniert auf einmal der touchscreen nicht mehr.
Auf dem Desktop und auch bei anderen Programmen funktioniert alles super.
Woran kann das liegen?

**Comment by AndreWohnsland** [comment]
Hallo Tuxrocks,

hast du über das all in one script installiert, bzw. kamen da irgendwelche Fehler?
Ich nehme an, dass du einen Raspberry Pi verwendest?
Kannst du mir einmal die Python Version (`python -V`) und das Betriebssystem (`cat /etc/os-release`) von deinem Pi mitteilen, bzw. am besten auch um welchen Touchscreen es sich handelt?
Hast du nach dem Programmstart kurz gewartet? Manchmal kann es sein, dass das Program wegen langsamen Internet und co. am Anfang kurz "hängt" das sollte aber im Sekundenbereich liegen.
Kannst du eine Maus anschließen bzw. hast du getestet, ob das Programm generell mit der Maus funktioniert, und nur nicht der Touchscreen?

Allgemein hatte ich noch nicht dass Problem, dass der Touchscreen nur nicht in dem Programm funktioniert hat. Ich hatte nur schon mal mit Touchscreens Probleme, welche in feuchten Umgebungen lagern und dann Teilweise einfach nicht reagieren, und dann auf einmal wieder, aber das ist dann unabhängig von dem Programm sondern in allen Fällen (also auch anderen Programmen) so.

Wenn des Terminal beim Starten von CocktailBerry irgendwelche Warnungen oder Fehler ausgibt, kannst du gerne davon auch einen Screenshot schicken, vielleicht lässt sich daraus auch ableiten, was das Problem ist.

Und als Nachtrag:
Die /boot/config.txt datei ist bestimmt auch nicht schlecht. Meistens ändert man (entweder selbst oder der touchscreen installer) da die config, es könnte ggf. auch an dem Treiber (dtoverlay) liegen der verwendet wird.

**Reply by Tuxrocks** [reply]
Hallo Andre,
da ich mir nicht sicher bin ob die direkten Mails angekommen sind und es wohl auch besser ist hier zu antworten um es für jeden zugänglich zu machen hier nochmal meine Antwort auf deine Fragen!

Hab das System mal komplett neu aufgesetzt da ich ja doch schon einiges versucht hab 


Ja -- habe über das all in one script installiert.  -> Warnmeldungen als Screenshot
Ja -- benutze einen Raspberry Pi
Python Version 3.9.2
OS : Debian GNU/Linux 11 (bullseye)
Touchscreen 5 Zoll : https://www.ebay.de/itm/274948720213?var=575228843081

Nach dem Programmstart ist hier und da ne kleine Wartezeit nötig aber mit der Maus ist alles wunderbar bedienbar.

Jetzt kommen auch schon mal keine Warnungen beim Start von Cocktailberry ->Screenshot

die config.txt:

```
# For more options and information see
# http://rpf.io/configtxt
# Some settings may impact device functionality. See link above for details

# uncomment if you get no picture on HDMI for a default "safe" mode
#hdmi_safe=1

# uncomment the following to adjust overscan. Use positive numbers if console
# goes off screen, and negative if there is too much border
#overscan_left=16
#overscan_right=16
#overscan_top=16
#overscan_bottom=16

# uncomment to force a console size. By default it will be display's size minus
# overscan.
#framebuffer_width=1280
#framebuffer_height=720

# uncomment if hdmi display is not detected and composite is being output
#hdmi_force_hotplug=1

# uncomment to force a specific HDMI mode (this will force VGA)
#hdmi_group=1
#hdmi_mode=1

# uncomment to force a HDMI mode rather than DVI. This can make audio work in
# DMT (computer monitor) modes
#hdmi_drive=2

# uncomment to increase signal to HDMI, if you have interference, blanking, or
# no display
#config_hdmi_boost=4

# uncomment for composite PAL
#sdtv_mode=2

#uncomment to overclock the arm. 700 MHz is the default.
#arm_freq=800

# Uncomment some or all of these to enable the optional hardware interfaces
#dtparam=i2c_arm=on
#dtparam=i2s=on
#dtparam=spi=on

# Uncomment this to enable infrared communication.
#dtoverlay=gpio-ir,gpio_pin=17
#dtoverlay=gpio-ir-tx,gpio_pin=18

# Additional overlays and parameters are documented /boot/overlays/README

# Enable audio (loads snd_bcm2835)
dtparam=audio=on

# Automatically load overlays for detected cameras
camera_auto_detect=1

# Automatically load overlays for detected DSI displays
display_auto_detect=1

# Enable DRM VC4 V3D driver
#dtoverlay=vc4-kms-v3d
max_framebuffers=2

# Run in 64-bit mode
arm_64bit=1

# Disable compensation for displays with overscan
disable_overscan=1

[cm4]
# Enable host mode on the 2711 built-in XHCI USB controller.
# This line should be removed if the legacy DWC2 controller is required
# (e.g. for USB device mode) or if USB support is not required.
otg_mode=1

[all]

[pi4]
# Run as fast as firmware / board allows
arm_boost=1

[all]
hdmi_force_hotplug=1
dtparam=i2c_arm=on
dtparam=spi=on
enable_uart=1
display_rotate=0
max_usb_current=1
config_hdmi_boost=7
hdmi_group=2
hdmi_mode=1
hdmi_mode=87
hdmi_drive=1
hdmi_cvt 800 480 60 6 0 0 0
dtoverlay=ads7846,cs=1,penirq=25,penirq_pull=2,speed=50000,keep_vref_on=0,swapxy=0,pmax=255,xohms=150,xmin=200,xmax=3900,ymin=200,ymax=3900
```

![IMG_5110](https://github.com/AndreWohnsland/CocktailBerry/assets/138435714/96e0191a-9c33-41f2-9f5f-a112ad9ce604)
![IMG_5111](https://github.com/AndreWohnsland/CocktailBerry/assets/138435714/a3367da0-124b-4ce4-b770-4f687092afea)
![IMG_5112](https://github.com/AndreWohnsland/CocktailBerry/assets/138435714/e0688db5-7727-4bd5-82ad-ef75228b2e1a)

hoffe du kannst mir helfen!

Vielen Dank im Vorraus
Ralf

**Reply by AndreWohnsland** [reply]
Hallo Ralf,

Hardware remote debuggen ist immer eine Kür, deshalb verzeihe, wenn auch ein paar dumme bzw. triviale Fragen dabei sind.

1. Wenn ich den Touchscreen mir so anschaue, nehme ich an, dass dieser ist über HDMI und alle Raspberry Pi Pins (über diesen "Plastikanschluss" auf der Touchrückseite) angeschlossen ist? Ansonsten würde ja auch generell der Touch nicht gehen.
2. War eine Installationsdatei, bzw. ein Script bei dem Touchscreen dabei, dass du ausgeführt hast? Der letzte Eintrag in der Config lässt jedenfalls vermuten, dass dieser geändert wurde:  `dtoverlay=ads7846,cs=1,penirq=25,penirq_pull=2,speed=50000,...`. Da stehen u.a. auch Kalibrierungswerte, die teilweise relativ hoch erscheinen (3900 für n 800er-Display, aber da bin ich nicht so drin).
3. Generell sieht das System, Python und Installation okay aus. Die Warnung kann man vernachlässigen, diese sollte keinen Einfluss haben. Das auch beim Start kein QT Fehler / Warnung kommt, stellt halt die Frage, warum der Touch nicht geht.

Was man noch probieren könnte:
- In den Optionen `UI_DEVENVIRONMENT` aktivieren, das zeigt ja den Cursor an. Dann auf dem Touch in der App in verschiedene Ecken klicken und schauen, ob sich der Cursor gar nicht, oder nur zufällig ein wenig wo hinbewegt. Wenn es das letztere ist, handelt es sich um ein Kalibrierungsproblem.
- evtest installieren (`sudo apt-get install evtest`) und dann in einem zweiten Terminal laufen lassen (`evtest`), damit kann man auch sehen, ob touchevents registriert werden, wenn die CocktailBerry App läuft.
- In den CocktailBerry Ordner gehen und erstmal `export QT_LOGGING_RULES="qt.qpa.input=true"` ausführen und dann `python runme.py` das Programm starten. Das habe ich noch nicht getestet, aber ggf. bekommt man dann beim Programm + Touch anklicken mehr Informationen in der Konsole.
- Wenn du immer mit dem `sh ~/launcher.sh` das Programm startest, in diesem einmal die Zeile `export QT_SCALE_FACTOR=1` auskommentieren, oder wegmachen. Ich glaube zwar nicht, dass das bei dir das Problem verursacht, aber lieber auf Nummer sicher sein.
- Generell scheint es ja erstmal nicht "direkt" an CocktailBerry, sondern an dem Grafik-Framework (Qt, hier die Python Adaption PyQt) zu handeln. Wenn alles keine Lösung bringt, hilft es ggf. auch in allgemeinen Forum dieses Problem mal anzusprechen. Ich denke du wirst nicht der erste sein, der dann Probleme mit Qt und Touchbildschirmen hat.

Ich hatte einmal eine vergleichbare 3,5-Zoll-Variante von dem Bildschirm für eine reine Touch Applikation, da ging Qt (PyQt) ohne Probleme.
Generell würde ich den Touch aufgrund seiner Bauweise (der Signalanschluss bedeckt alle Pins, wie will man dann andere Sachen anschließen / steuern) nicht für Projekte empfehlen, die mehr als "nur" einen Touchscreen bereitstellen (also noch Maschinen, Sensoren, etc. angeschlossen sind). 
Leider ist es echt das erste Mal, dass sowas auftritt. Diverse 5- und 7-Zoll Monitore von mir gingen bis jetzt, auch die von anderen Nutzern hatten bis jetzt nicht so ein Problem. Auch mit dem "offiziellen" 7-Zoll Raspberry Pi Touchscreen läuft die Software problemlos.

PS: Mails kamen keine, für E-Mail-Kommunikationen (sensitive Sachen, die man nicht im Internet will) hatte ich einmal cocktailmakeraw@gmail.com eingerichtet. Hier im Diskussionsforum ist generell aber auch ein guter Platz für so Probleme.

Grüße
Andre

**Reply by Tuxrocks** [reply]
Hi nochmal,

1. Ja der Touch steckt im Moment über den GPIO‘s. Hab mir den geholt weil man mit ein paar Kabeln (glaube so ca 8 Pins) das Display auch extern betreiben kann. Hab ich vor ca einem Jahr in nem anderen Projekt umgesetzt— da muss der Touch allerdings nur ne HTML Seite bedienen was auch funktioniert 
2. Ja hab ne Installationsdatei ausgeführt. Einmal mit der auf CD beiliegenden und ein anderes Mal mit der aus dem Web. (LCD Show 5) glaub ich. Beides ohne Erfolg
3. Das dachte ich mir auch — da ja „nur Warnungen“ und keine Fehler kamen!

Versucht habe ich:

UI_DEVENVIRONMENT : keine Reaktion 

evtest reagiert auf dem Desktop jedesmal wenn ich was antippe - in Cocktailberry keine Reaktion bzw Meldung

export QT_LOGGING_RULES="qt.qpa.input=true" —
keine Meldungen 

export QT_SCALE_FACTOR=1 — keine Änderung 

hab gestern auch mal kurz gegoogelt leider werd ich daraus auch nicht schlau 🙈


Hab jetzt mal das 7“ aus Verzweiflung geordert wär aber trotzdem klasse wenn’s noch zu einer Lösung mit dem 5“ käm 
Schlecht wär dann wenn das 7“ auch nicht funktioniert 😀

Nachtrag:
hab jetzt mal n paar Sachen ausprobiert und Fehler beim Start erzeugt — evtl helfen die ja 🤷🏼‍♂️
![FE81265A-3D5B-42B3-ADA5-441A2847E50B](https://github.com/AndreWohnsland/CocktailBerry/assets/138435714/b5a73f88-35a1-4971-940a-e5ddeb46ed3a)

**Reply by Tuxrocks** [reply]
Und noch was ist mir aufgefallen!
Das Paket sudo apt-get install qt5-default
gibt es anscheinend nicht mehr.
Evtl is da ja der Haken ?

**Reply by AndreWohnsland** [reply]
Hi, 

# Ersmal zum Orginal Post

Ja diese ganzen diversen Protokollen, bzw. Schnittstellen, die die HMI Bildschirme verwenden können manchmal verwirrend sein.

Was man bei dem 5 Zoll noch versuchen könnte ist (wenn ich es richtig gesehen habe hat der doch auch einen Microusb Anschluss?) diesen über den Microusb anschluss an den USB vom Pi zu schließen (falls das angesehen ist für Signalübertragung). Viele der anderen Touch haben das als Standartverbindung, vielleicht kann das dieser auch, da muss man aber mal die Dokumentation von dem Hersteller anschauen (wenn die welche bereitstellen, das ist bei den Touchscreens auch so ein Ding....).

So wie es scheint kann Qt echt einfach nicht das Signal auslesen, bzw. das Signal der `ads7846` Schnittstelle vom Touch kommt nicht in Qt an. Ob das durch ein neucompilieren von Qt auf diese Schnitstelle, oder durch tiefere Analysen, ob generell das Verzeichniss, wo das Signal herkommt einfach anders ist, ist halt echt schwierig "schnell" und remote herauszufinden 😅. Qt bekommt das in der Regel von `/dev/input/`, aber wie man das genau dann alles debugged kann ich dir leider auch nicht sagen, so tief bin ich dann auch nicht da drinne. Laut den Tests und deinen Antworten kann man jedenfalls sagen, dass Qt keine Touch-Inputs registriert.

Ich hoffe (und vermute ja mal) dass es mit dem 7er geht, plus ich finde auch dass die 7 Zoll deutlich schöner für die Bedienung als 5 sind. Ich fand ich auch die Einrichtung von meinen 7ern deutlich einfacher (Teilweise muss man eine Zeile ändern und das wars.). Habe meine alten mit 5 auch vor einiger Zeit auf 7 geupgraded. Welchen 7-Zoll hast du dir jetzt bestellt? Und im Notfall findet sich immer ein anderes Projekt für existierende Hardware (den 5er Zoll 😉)

# Dann noch zum Edit und 2. Post

Die Warnung hatte bzw. hab ich auch bei ein Paar Setups, das sollte nicht das Problem sein.
Die andere Meldung muss ich mir erstmal anschauen. 
Habe da auch gesehen, dass du da das 32 Bit Betriebssystem getestet hast - vorher wars ja das 64er, was ich üblicherweise auch empfehle.

Wegen dem qt5-default, die kurze Antwort: Qt ist auf jedenfall installiert, sonst würde die Applikation nicht gehen.
Die etwas längere: `qt5-default` ist noch für Raspbian Buster, also das Debian 10 (Derzeit ist 11, bald sogar 12 der Standard), das [installer Script](https://github.com/AndreWohnsland/CocktailBerry/blob/master/scripts/setup.sh#L53) versucht erst diese zu installieren, fällt wenn es nicht klappt auf `python3-pyqt5` zurück. Das wird derzeit üblicherweise installiert und damit läuft die App dann. Denn mit Debian 11 gibt es kein qt5-default paket mehr.

Was man da aber noch versuchen könnte:
- Deinstallieren von python3-pyqt5 (`sudo apt remove python3-pyqt5`)
- Über Pythons Pip versuchen, das braucht manchmal bzw. geht manchmal auch nicht. Es sollte theoretisch gehen mit 
 - `sudo apt install qtbase5-dev` <- das muss vor dem pip installiert werden, sonst wirft der pip wahrscheinlich Fehler
 - `pip install PyQt5`
 
Alternativ könnte man noch die äquivalenten qt-Pakete mit `sudo apt install qtbase5-dev qtchooser qt5-qmake qtbase5-dev-tools` versuchen zu installieren. Angeblich soll das ziemlich das identische enthalten.

Grüße

## Und noch ein Nachtrag:

Welche Variante von dem Pi ist es? Ein RPi3B oder RPi3B+, oder doch das 4er?

**Reply by Tuxrocks** [reply, marked as answer]
Alsoooo 😀
Mit dem „Original“ 7“ funktioniert alles auf Anhieb! Danke dafür schon mal.
Hab das Ganze am 3B+ als auch am 4er versucht — gleiches Verhalten!
Alles andere hat bis jetzt zu keinem Erfolg geführt! 
ich bleib an der Sache dran — evtl findet sich ja noch die Lösung!
Gruß und danke für deine Mühen bis jetzt.
Evtl muss ich dir ja nochmal auf die Nerven gehen 😜
Ralf

**Reply by AndreWohnsland** [reply]
Okay, dann scheint das echt ein Problem mit Qt und dieser speziellen HMI Monitor Schnittstelle zu sein. Freut mich, dass es immerhin mit dem anderen klappt 😄 
Falls sich eine Lösung für den 5-zoller ergibt, bin ich gern dran interessiert.

**Reply by Tuxrocks** [reply]
Hi wieder mal!
Also — lang ists her und für den 5-zoller ist keine Lösung gefunden.
Hab alles versucht was für mich machbar war.
Jetzt hat er seine Bestimmung in der Hausautomation gefunden da er so nur n paar Buttons auf ner Website bedienen muss was er auch macht 😊
Danke nochmal für deine Hilfe

**Reply by AndreWohnsland** [reply]
Hi, schön zu hören, dass der andere Bildschirm einen Zweck gefunden hat. Doof dass es nicht ging, aber das muss etwas mit dem unterliegenden Grafik Ressourcen zu tun haben, da hat man dann leider wenig Einfluss.
